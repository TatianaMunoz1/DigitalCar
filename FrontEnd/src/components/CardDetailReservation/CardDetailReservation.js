import React from "react";
import { useHistory } from "react-router-dom";
import styles from "../CardDetailReservation/CardDetailReservation.module.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faMapMarkerAlt } from "@fortawesome/free-solid-svg-icons";
import Rating from "../Rating/Rating";
import Button from "../Button/Button";
import Url from "../../Url"
import axios from "axios";
import Swal from "sweetalert2";

function CardDetailReservation(props) {
  const car = props.car;
  const dateReservation = props.dateReservation;
  const userCityInput = props.userCityInput;
  const history = useHistory();

  const onSubmit = (e) => {
    e.preventDefault();

    if (
      userCityInput === null ||
      dateReservation.startTime === null ||
      dateReservation.endDate === null ||
      dateReservation.startDate === null
    ) {
      props.handleValidation(false);
    } else {
      const config = {
        headers: {
          Authorization: props.user.token,
        },
      };

      const body = {
        car: { id: car.id },
        startDateTime:
          dateReservation.startDate.toISOString().slice(0, 11) +
          dateReservation.startTime.toISOString().slice(11, 21),
        endDate: dateReservation.endDate.toISOString().slice(0, 10),
        user: { email: props.user.email },
      };

      axios
        .post(Url+"/booking", body, config)
        .then(function (response) {
          if (response.status === 201) {
            history.push("/succes-reservation");
          } else {
            const Toast = Swal.mixin({
              toast: true,
              position: "top-end",
              showConfirmButton: false,
              background: "#ffc2c2",
              timer: 3000,
              timerProgressBar: true,
              didOpen: (toast) => {
                toast.addEventListener("mouseenter", Swal.stopTimer);
                toast.addEventListener("mouseleave", Swal.resumeTimer);
              },
            });
            Toast.fire({
              icon: "error",
              title:
                "<h4 style='color:#B00020'>Para realizar una reserva necesitas estar logueado</h4>",
            });
          }
        })
        .catch(function (error) {
          const Toast = Swal.mixin({
            toast: true,
            position: "top-end",
            showConfirmButton: false,
            background: "#ffc2c2",
            timer: 3000,
            timerProgressBar: true,
            didOpen: (toast) => {
              toast.addEventListener("mouseenter", Swal.stopTimer);
              toast.addEventListener("mouseleave", Swal.resumeTimer);
            },
          });
          Toast.fire({
            icon: "error",
            title:
              "<h4 style='color:#B00020'>No se pudo procesar su solicitud</h4>",
          });
        });
    }
  };

  return car ? (
    <div className={styles.container}>
      <h1 className={styles.title}>Detalle de la reserva</h1>
      <div className={styles.container_information}>
        <div className={styles.image_container}>
          <img
            src={car.images[0].url}
            alt={car.title}
            className={styles.image}
          />
        </div>
        <form className={styles.information} onSubmit={onSubmit}>
          <div className={styles.datails_box}>
            <div className={styles.datails}>
              <h4 className={styles.category}>
                {car.category.title.toUpperCase()}
              </h4>
              <h1 className={styles.carName}>{car.name}</h1>
              <Rating dontShowLabel={true} dontShowRating={true} />
              <h4 className={styles.subtitle_location}>
                <FontAwesomeIcon
                  icon={faMapMarkerAlt}
                  className={styles.mapBtn}
                />
                &nbsp;
                {car.address}
                {car.city.name}, {car.city.country}
              </h4>
            </div>
            <div className={styles.dates_box}>
              <p className={styles.dates}>
                <span>Check in</span>
                {props.dateReservation.startDate !== null ? (
                  <span className={styles.date_start}>
                    {props.dateReservation.startDate
                      .toLocaleDateString("es")
                      .toString()}
                  </span>
                ) : (
                  <span className={styles.date_space}>_/_/_</span>
                )}
              </p>
              <p className={styles.dates}>
                <span>Check out</span>
                {props.dateReservation.endDate !== null ? (
                  <span className={styles.date_end}>
                    {props.dateReservation.endDate
                      .toLocaleDateString("es")
                      .toString()}
                  </span>
                ) : (
                  <span className={styles.date_space}>_/_/_</span>
                )}
              </p>
            </div>
          </div>
          <div className={styles.btn_reservation}>
            <Button
              title="Confirmar reserva"
              link={`/${car.id}/reservation`}
              type="submit"
              btnStyle="btn--full"
              btnSize="btn--responsive"
              onClick={onSubmit}
            />
          </div>
        </form>
      </div>
    </div>
  ) : (
    <></>
  );
}

export default CardDetailReservation;
