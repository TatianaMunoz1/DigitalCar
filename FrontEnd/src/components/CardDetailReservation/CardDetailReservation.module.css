* {
  box-sizing: border-box;
  margin: 0%;
}
.container {
  width: 100%;
  height: 100%;
  background: #ffffff;
  box-sizing: border-box;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 8px;
}
.title:first-child {
  padding: 20px;
}
.category {
  color: rgb(146, 146, 146);
}
.title {
  color: #000;
  font-size: 1.6rem;
}
.carName {
  font-size: 1.6rem;
  line-height: 1.3rem;
  padding-top: 5px;
}
.subtitle_location {
  padding: 10px 0px;
  font-weight: 500;
}
.image_container {
  width: 100%;
  height: 300px;
  box-sizing: border-box;
  filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
}
.image {
  width: 100%;
  height: 100%;
  object-fit: cover;
}
.information {
  padding: 15px;
  height: auto;
}
.datails_box {
  display: flex;
  flex-direction: column;
  gap: 40px;
  padding-bottom: 30px;
}
.dates {
  padding: 20px 0px;
  display: flex;
  justify-content: space-between;
}
.dates:first-child {
  border-top: 1px solid #bebebe;
}
.dates:last-child {
  border-top: 1px solid #bebebe;
  border-bottom: 1px solid #bebebe;
}
.date_space {
  transform: scale(1.6, 1);
  margin-right: 15px;
}

/*-----TABLET-----*/
@media (min-width: 760px) and (max-width: 1280px) {
  .container {
    width: 700px;
    padding: 20px;
  }
  .title:first-child {
    padding: 0px 0px 10px 0px;
  }
  .container_information {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 15px;
  }

  .image_container {
    width: 100%;
    height: 100%;
    object-fit: cover;
    border: 1px solid #000000;
    border-radius: 5px;
  }
  .image {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
  .information {
    padding: 0px;
  }
}

/*-----DESKTOP-----*/
@media (min-width: 1281px) {
  .image_container {
    width: 100%;
    height: 350px;
  }
  .information {
    padding: 20px;
  }
  .container {
    position: relative;
  }
  .datails_box {
    display: flex;
    flex-direction: column;
    gap: 40px;
  }
  .btn_reservation {
    position: absolute;
    width: 100%;
    padding: 20px;
    bottom: 0px;
    left: 0px;
  }
}
