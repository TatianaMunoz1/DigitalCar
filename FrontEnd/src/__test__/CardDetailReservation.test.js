import React, { useEffect } from "react";
import { shallow, mount, render, mock } from "enzyme";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import CardDetailReservation from "../components/CardDetailReservation/CardDetailReservation.js";
const carTest = {
    id: 1,
    title: "CitroÃ«n C4",
    seats: 5,
    doors: 5,
    transmission: "Manual",
    baggage: 2,
    description:
      "Esta oferta incluye 100km gratis! The fire truck braked the upfitted Express 1500. The mechanical Hino was developed by the Hino and once the tow truck totaled the seat belt.",
    category: "SUV",
    features: [{ name: "name", details: "details" }],
    img: "https://www.sixt.com.ar/fileadmin/files/global/user_upload/fleet/png/350x200/citroen-c4-5d-blue-2021.png",
    handleFav: () => {},
  };
  
describe("Create detail reservation", () => {
    const history = createMemoryHistory();
  test("must img exist", () => {
    const wrapper = mount(
      <Router history={history}>
        <CardDetailReservation {...carTest}/>
      </Router>
    );

    //expect(wrapper.find("img").prop("src")).toEqual(car.img);
    expect(wrapper.text().includes("Detalle de la reserva")).toBeTruthy();
  });

    test("render card detail reservation ", () => {
        const wrapper = shallow(<CardDetailReservation/>)
        expect(wrapper).toMatchSnapshot();
    });
});