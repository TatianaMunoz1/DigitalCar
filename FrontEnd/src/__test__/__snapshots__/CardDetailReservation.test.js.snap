// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create detail reservation render card detail reservation  1`] = `ShallowWrapper {}`;
